controllers/datadogagent/secret_agent.go

```diff
@@ -17,26 +15,21 @@ import (
 	apiutils "github.com/DataDog/datadog-operator/apis/utils"
 	"github.com/DataDog/datadog-operator/pkg/config"
 	"github.com/DataDog/datadog-operator/pkg/controller/utils"
-	"github.com/go-logr/logr"
+	"github.com/DataDog/datadog-operator/pkg/secrets"
 )
 
 func (r *Reconciler) manageAgentSecret(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, newStatus *datadoghqv1alpha1.DatadogAgentStatus) (reconcile.Result, error) {
 	return r.manageSecret(logger, managedSecret{name: utils.GetDefaultCredentialsSecretName(dda), requireFunc: needAgentSecret, createFunc: newAgentSecret}, dda, newStatus)
 }
 
 func newAgentSecret(name string, dda *datadoghqv1alpha1.DatadogAgent) (*corev1.Secret, error) {
-	if err := checkCredentials(dda); err != nil {
-		return nil, err
-	}
-
```

Comment:
I think we still need to check that the API (and maybe APP) keys are provided as envvar or in the DatadogAgent.spec

Reply:
yes, it is checked in the `needAgentSecret` function, which is required to be truthy before `newAgentSecret` is called

Reply:
i thought about that more, and i think the conditions to check whether the secret needs to be created are checked for all accounted for in the `sufficiency` functions. they're a combination of checking for spec credential existence, checking if it uses secret backend, and if there is an envvar fallback. happy to discuss more (perhaps IRL)